src/generic/grideditors.cpp

```diff
@@ -1710,5 +1711,107 @@ wxGridCellAutoWrapStringEditor::Create(wxWindow* parent,
                                     wxTE_MULTILINE | wxTE_RICH);
 }
 
+#if wxUSE_DATEPICKCTRL
+
+// ----------------------------------------------------------------------------
+// wxGridCellDateEditor
+// ----------------------------------------------------------------------------
+
+wxGridCellDateEditor::wxGridCellDateEditor()
+{
+}
+
+void wxGridCellDateEditor::Create(wxWindow* parent, wxWindowID id,
+                                  wxEvtHandler* evtHandler)
+{
+    m_control = new wxDatePickerCtrl(parent, id);
+
+    wxGridCellEditor::Create(parent, id, evtHandler);
+}
+
+void wxGridCellDateEditor::SetSize(const wxRect& r)
+{
+    wxRect rect = r;
+    wxSize bestSize = m_control->GetBestSize();
+    bool resize = false;
+
+    if( rect.GetHeight() < bestSize.GetHeight() )
```

Comment:
This could be replaced by a simpler and more readable call to `wxSize::IncTo()`.

It doesn't return if the size has changed or not, but as we call `m_control->SetSize()` unconditionally anyhow, this shouldn't be a problem.